### PR TITLE
New Parse EmailHeaders script

### PR DESCRIPTION
**Pull request by dem-evgeny**

### CODE REVIEW

File: Scripts/script-ParseEmailHeaders.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  # Parse the headers from the email file
+  parser = HeaderParser()
+  h = parser.parsestr(filecontents)
+
+  # Print all the headers and add them to the context
+  for key in h.keys():
+      md += key + " : " + h[key] + "\n"
+      ec["Email.Headers"][key] = h[key]
+
+  demisto.results({"ContentsFormat": formats["json"], "Type": entryTypes["note"], "Contents": h.keys(), "HumanReadable": md, "EntryContext": ec})
```
Comment (gshriki) [inline Scripts/script-ParseEmailHeaders.yml]: no outputs defined?

File: Scripts/script-ParseEmailHeaders.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+type: python
+tags:
+- email
+comment: This automation parses headers from an email file.
+enabled: true
+args:
+- name: entryid
```
Comment (gshriki) [inline Scripts/script-ParseEmailHeaders.yml]: entryID

File: Scripts/script-ParseEmailHeaders.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  required: true
+  default: true
+  description: Entry id of an email file
```
Comment (gshriki) [inline Scripts/script-ParseEmailHeaders.yml]: id -> ID

File: Scripts/script-ParseEmailHeaders.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  version: -1
+name: ParseEmailHeaders
+script: |-
+  from email.parser import HeaderParser
+
+  ec = {"Email.Headers" : {}}
+  md = ""
+
+  # Get the file path of the email file as it lokated in Demisto server
+  filePath = demisto.executeCommand('getFilePath', {'id': demisto.args()['entryid']})[0]['Contents']['path']
```
Comment (gshriki) [inline Scripts/script-ParseEmailHeaders.yml]: Did you check all scenarios? What if file does not exist? what if it is not an email file?

File: Scripts/script-ParseEmailHeaders.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+commonfields:
+  id: ParseEmailHeaders
+  version: -1
+name: ParseEmailHeaders
+script: |-
+  from email.parser import HeaderParser
+
+  ec = {"Email.Headers" : {}}
```
Comment (gshriki) [inline Scripts/script-ParseEmailHeaders.yml]: `"` -> `'`

File: Scripts/script-ParseEmailHeaders.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  md = ""
+
+  # Get the file path of the email file as it lokated in Demisto server
+  filePath = demisto.executeCommand('getFilePath', {'id': demisto.args()['entryid']})[0]['Contents']['path']
+
+  # Read the content of the file
+  f = open(filePath, 'rb')
+  filecontents = f.read()
```
Comment (gshriki) [inline Scripts/script-ParseEmailHeaders.yml]: fileContents

File: Scripts/script-ParseEmailHeaders.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+  # Parse the headers from the email file
+  parser = HeaderParser()
+  h = parser.parsestr(filecontents)
+
+  # Print all the headers and add them to the context
+  for key in h.keys():
+      md += key + " : " + h[key] + "\n"
```
Comment (gshriki) [inline Scripts/script-ParseEmailHeaders.yml]: `"` -> `'`

File: Scripts/script-ParseEmailHeaders.yml (hunk 1 above)
Comment (gshriki) [inline Scripts/script-ParseEmailHeaders.yml]: `"` -> `'`

File: Scripts/script-ParseEmailHeaders.yml (hunk 5 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: fixed

Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: Fixed back.
In Python it should be " and not '

File: Scripts/script-ParseEmailHeaders.yml (hunk 4 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: Fixed

File: Scripts/script-ParseEmailHeaders.yml (hunk 6 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: fixed

File: Scripts/script-ParseEmailHeaders.yml (hunk 7 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: It is Python not JS

File: Scripts/script-ParseEmailHeaders.yml (hunk 1 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: Added

Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: It is Python not JS

File: Scripts/script-ParseEmailHeaders.yml (hunk 2 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: Fixed

File: Scripts/script-ParseEmailHeaders.yml (hunk 3 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: Fixed

File: Scripts/script-ParseEmailHeaders.yml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  from email.parser import HeaderParser
+
+  ec = {"Email.Headers" : {}}
+  md = ""
+
+  # Get the file path of the email file as it lokated in Demisto server
+  filePath = demisto.executeCommand('getFilePath', {'id': demisto.args()['entryid']})[0]['Contents']['path']
+
+  # Read the content of the file
+  f = open(filePath, 'rb')
```
Comment (orius123) [inline Scripts/script-ParseEmailHeaders.yml]: You better use `with open((filePath, 'rb') as f`...
or at least close the file afterwards

Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: Changed

File: Scripts/script-ParseEmailHeaders.yml [hunk 9]
```diff
@@ -0,0 +1,46 @@
+commonfields:
+  id: ParseEmailHeaders
+  version: 34
```
Comment (orius123) [inline Scripts/script-ParseEmailHeaders.yml]: version should be -1

File: Scripts/script-ParseEmailHeaders.yml [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+          # Parse the headers from the email file
+          parser = HeaderParser()
+          h = parser.parsestr(fileContents)
+
+          # Print all the headers and add them to the context
+          for key in h.keys():
+              md += key + " : " + h[key] + "\n"
+              ec["Email.Headers"][key] = h[key]
+
+          demisto.results({"ContentsFormat": formats["json"], "Type": entryTypes["note"], "Contents": h.keys(), "HumanReadable": md, "EntryContext": ec})
```
Comment (orius123) [inline Scripts/script-ParseEmailHeaders.yml]: Shouldn't format be markdown?

File: Scripts/script-ParseEmailHeaders.yml [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+      else:
+          demisto.results("This file is not exists or it is not eml file type\n");
```
Comment (orius123) [inline Scripts/script-ParseEmailHeaders.yml]: I think this message is better - `"Couldn't find EML file for entry ID: '" + demisto.args()['entryId'] + "'"`

File: Scripts/script-ParseEmailHeaders.yml (hunk 9 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: changed

File: Scripts/script-ParseEmailHeaders.yml (hunk 10 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: I looked at other integrations and no.

File: Scripts/script-ParseEmailHeaders.yml (hunk 11 above)
Comment (dem-evgeny) [reply Scripts/script-ParseEmailHeaders.yml]: Changed

**Pull request merged**